docs/_ext/validating_code_block.py

```diff
@@ -0,0 +1,63 @@
+from typing import List
+from docutils import nodes
+from docutils.parsers.rst import Directive
+from docutils.parsers.rst import directives
+from sphinx.application import Sphinx
+from sphinx.util.docutils import SphinxDirective
+from sphinx.directives.code import CodeBlock
+from sphinx.errors import ExtensionError
+
+import os
+import subprocess
+
+
+class ValidatingCodeBlock(CodeBlock):
+  """A directive that provides protobuf yaml formatting and validation.
+
+    'type-name' option is required and expected to conain full Envoy API type.
+    An ExtensionError is raised on validation failure.
+    Validation will be skipped if SPHINX_SKIP_CONFIG_VALIDATION environment variable is set.
+    """
+  has_content = True
+  required_arguments = CodeBlock.required_arguments
+  optional_arguments = CodeBlock.optional_arguments
+  final_argument_whitespace = CodeBlock.final_argument_whitespace
+  option_spec = {
+      'type-name': directives.unchanged,
+  }
+  option_spec.update(CodeBlock.option_spec)
+  skip_validation = os.getenv('SPHINX_SKIP_CONFIG_VALIDATION') or 'false'
+  bazel_build_options = os.getenv('BAZEL_BUILD_OPTIONS') or ''
+
+  def run(self):
+    source, line = self.state_machine.get_source_and_line(self.lineno)
+    # built-in directives.unchanged_required option validator produces a confusing error message
+    if self.options.get('type-name') == None:
+      raise ExtensionError("Expected type name in: {0} line: {1}".format(source, line))
+
+    if ValidatingCodeBlock.skip_validation.lower() != 'true':
```

Comment:
`if bool(ValidatingCodeblock.skip_validation()):` (or something like this, i.e. you can parse the bool).

Reply:
Hmm:
```
>>> bool("false")
True
```

Reply:
OK, forget it :)